### PR TITLE
feat: added rtl direction to all of ant-design components

### LINKED ISSUE #4051: RTL support all pages and components?
Is ant support RTL languages?
Persian, Arabic and etc. ..

### PR DESCRIPTION
**Pull request by saeedrahimi**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close #4051

Previous pull request was on `feature` branch: #18006 
### 💡 Background and solution

As requirement of this [issue](https://github.com/ant-design/ant-design/issues/4051) components of ant-design doesn't support right to left direction, and those components are not useful in some languages like Persian(Farsi) or Arabic.

Solution: 
I started to develop Rtl style on ant-design and as @zombieJ  mentioned in issue i used ConfigProvider to set direction of component.

direction property added to ConfigProvider.
```
<ConfigProvider direction="rtl">
      <Cascader  options={options} placeholder="select an item"/>
</ConfigProvider>
```
in style files added prefixed rtl styles to change direction of displayed items
```
  .@{inputClass}-suffix {
    right: @input-padding-horizontal-base + 1px;

    .@{inputClass}-rtl & {
      right: auto;
      left: @input-padding-horizontal-base + 1px;
    }
  }
```
so when RTL'ed components wrapped by ConfigProvider with `direction='rtl'` property, all child components (also grid-system) will shown in right-to-left direction.
ant-rtl

## RTL Progress

- [x] Button
- [x] Icon
- [x] Typography
- [x] Grid
- [x] Layout
- [x] Affix
- [x] Breadcrumb
- [x] Dropdown
- [x] Menu
- [x] Pagination
- [x] PageHeader
- [x] Steps
- [x] AutoComplete
- [x] CheckBox
- [x] Cascader
- [x] DatePicker
- [x] Form
- [x] InputNumber
- [x] Input
- [x] Mention
- [x] Rate
- [x] Radio
- [x] Switch
- [x] Slider
- [x] Select
- [x] TreeSelect
- [x] Transfer
- [x] TimePicker
- [x] Upload
- [x] Avatar
- [x] Badge
- [x] Comment
- [x] Collapse
- [x] Carousel
- [x] Cad
- [x] Calendar
- [x] Description List
- [x] Empty
- [x] List
- [x] Popover
- [x] Statistic
- [x] Tree
- [x] Tooltip
- [x] Timeline
- [x] Tag
- [x] Tabs (Done)
- [x] Table
- [x] Alert
- [x] Drawer
- [x] Modal
- [ ] Message
- [ ] Notification
- [x] Progress
- [x] Popconfirm
- [x] Result
- [x] Spin
- [x] Skeleton
- [x] Anchor (No need to RTL)
- [x] BackTop
- [x] ConfigProvider
- [x] Divider (No need to RTL)
- [x] Front page

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

-----
[View rendered components/config-provider/demo/direction.md](https://github.com/saeedrahimi/ant-design/blob/ant-rtl/components/config-provider/demo/direction.md)
[View rendered components/config-provider/index.en-US.md](https://github.com/saeedrahimi/ant-design/blob/ant-rtl/components/config-provider/index.en-US.md)
[View rendered components/config-provider/index.zh-CN.md](https://github.com/saeedrahimi/ant-design/blob/ant-rtl/components/config-provider/index.zh-CN.md)

### CODE REVIEW

**Comment by saeedrahimi** [comment]
I have no idea why lint fails, there is no error on my environment.
Windows 10 v17134.1069 
@afc163 , @zombieJ

**Comment by zombieJ** [comment]
`@typescript-eslint/parser` failed with markdown at `2.5.0`. It's OK if not other lint failed without markdown file:

ref https://github.com/typescript-eslint/typescript-eslint/issues/1123

**Comment by mahdi13** [comment]
@saeedrahimi I check this page every day! Kindly ask you to finish it.

**Comment by zombieJ** [comment]
@saeedrahimi,
`4.0-prepare` synced with `master`. Some conflict need rebase `4.0-prepare`. Thanks.

**Comment by saeedrahimi** [comment]
@afc163 , @zombieJ 
Hi there,
ant-design-rtl-button

I thought that config-provider demo page is not suitable to demonstrate all `rtl` components and their functionality in rtl direction.
so i added a button to header that changes direction of the whole demo.
I just wanted to know your idea about these changes. is it ok to continue with this?
I wanna do my best.
Thanks.

**Comment by yoyo837** [comment]
Great work, I hope that the owner can pay enough attention to this and like your contribution.

**Comment by zombieJ** [comment]
> @afc163 , @zombieJ
> Hi there,
> ant-design-rtl-button
> 
> I thought that config-provider demo page is not suitable to demonstrate all `rtl` components and their functionality in rtl direction.
> so i added a button to header that changes direction of the whole demo and updated some styles of `site/theme` to support rtl properly.
> I just wanted to know your idea about these changes. is it ok to continue with this?
> I wanna do my best.
> Thanks.

Cool. I'm OK with this if all the components support RTL.

**Comment by yoyo837** [comment]
Note that some components have been refactored to a new major version in `4.0-prepare` branch. And there are other components that are also planned to be refactored.

**Comment by saeedrahimi** [comment]
> Note that some components have been refactored to a new major version in `4.0-prepare` branch. And there are other components that are also planned to be refactored.

thanks for note, i will double check them to ensure components working correctly.

**Comment by saeedrahimi** [comment]
@zombieJ , I'm almost done with implementing rtl view of components.
Would you please review this pull request?
( about rtl of `message` and `notification` component i want to open an issue)

**Comment by shayegh** [comment]
سلام
خدا قوت 
انشاالله به زودی نهائی بشه و بتونیم ازش استفاده کنیم

**Comment by zombieJ** [comment]
hi @saeedrahimi , v4 plan release beta version on December 1. Is that possible to finish this feature at that time?

**Comment by saeedrahimi** [comment]
> @zombieJ , I'm almost done with implementing rtl view of components.
> Would you please review this pull request?
> ( about rtl of `message` and `notification` component i want to open an issue)

> hi @saeedrahimi , v4 plan release beta version on December 1. Is that possible to finish this feature at that time?
Yes,
I'm almost done, for now all components and front-page supports rtl direction. except `message` and `notification` components that i can't access `ConfigContext` in them,

**Comment by saeedrahimi** [comment]
> @zombieJ , I'm almost done with implementing rtl view of components.
> Would you please review this pull request?
> ( about rtl of `message` and `notification` component i want to open an issue)
 
> hi @saeedrahimi , v4 plan release beta version on December 1. Is that possible to finish this feature at that time?

Yes,
I'm almost done, for now all components and front-page supports rtl direction. except `message` and `notification` components that i can't access `ConfigContext` in them,

**Comment by zombieJ** [comment]
It's OK to leave them currently since it do not have content.

**Comment by saeedrahimi** [comment]
@zombieJ , i'm getting different snapshots in my environment. would you guide me how to resolve failing tests.?
Windows 10
npm 6.10.2

**Comment by zombieJ** [comment]
Could you attach the screenshot about error snapshot?

**Comment by saeedrahimi** [comment]
> Could you attach the screenshot about error snapshot?

I don't have any error in my environment, all tests are passing successfully in my environment.

**Comment by saeedrahimi** [comment]
CI tests are failing because of diff in snapshots. and i don't know why.
I've updated snapshots and still fails.

I checked some of tests, in `ci/circleci: test_dom` :
```

FAIL  components/locale-provider/__tests__/index.test.js (15.979s)
  Locale Provider
...
...
    ✕ should display the text as id (209ms)
...
    mount and unmount
      ✓ component could be updated and unmounted without errors (9ms)

  ● Locale Provider › should display the text as id

    expect(received).toMatchSnapshot()

    Snapshot name: `Locale Provider should display the text as id 1`

    - Snapshot
    + Received

    @@ -331,15 +331,15 @@
                            2017
                          </a>
                        </span>
                        <a
                          class="ant-calendar-next-month-btn"
    -                     title="Bulan depan (PageDown)"
    +                     title="Bulan selanjutnya (PageDown)"
                        />

```
 -   title="Bulan depan (PageDown)" is in  `rc-calendar` version `"9.15.5"` [link](https://github.com/react-component/calendar/blob/d486704af3357741bb150628d206a956a5d1cd61/src/locale/id_ID.js#L21)

+ title="Bulan selanjutnya (PageDown)" is in  `rc-calendar` version `"9.15.7+"` [link](https://github.com/react-component/calendar/blob/0f2c66f618fddf8cd563d3930a77abd8dcce39e5/src/locale/id_ID.js#L21)

and current `rc-calendar` version in `ant-design:4.0-prepare` is "~9.15.5"

**Comment by yoyo837** [comment]
Try delete node_modules and reinstall it.

**Comment by yoyo837** [comment]
image

**Comment by saeedrahimi** [comment]
After 4 hours of trying hard, finally i could run `npm install` successfully !

p.s: after protests in Iran we got internet black-out, we have internet now but with poor quality!

**Comment by zombieJ** [comment]
@saeedrahimi , could you help to check `rc-picker` if should update to support RTL? It's new DatePicker in v4
ref: https://github.com/react-component/picker

**Comment by zombieJ** [comment]
> After 4 hours of trying hard, finally i could run `npm install` successfully !
> 
> p.s: after protests in Iran we got internet black-out, we have internet now but with poor quality!

Thanks for your effort!  🌸

**Comment by saeedrahimi** [comment]
> 
> 
> @saeedrahimi , could you help to check `rc-picker` if should update to support RTL? It's new DatePicker in v4
> ref: https://github.com/react-component/picker

Yes, i would, but i can't deliver it before December 1 for v4 release.

**Comment by zombieJ** [comment]
> Yes, i would, but i can't deliver it before December 1 for v4 release.

That's OK. Thanks for your lot lot contribution : )

**Comment by saeedrahimi** [comment]
Hi @zombieJ ,
Is this pull request planned to merge with v4 release?
I'm done with this pull request and every time base branch changes, the process of resolving conflicts takes long time.

**Comment by yoyo837** [comment]
Let's merge this PR... 🚀

**Comment by saeedrahimi** [comment]
> Let's merge this PR... 🚀
Conflicts resolved ;)

Note this commit (link)[https://github.com/ant-design/ant-design/pull/19380/commits/3ba281d2e8dd5923eab0c9280ab0b2c024f105b1]
I deleted some workflows to allow push my codes.

**Comment by yoyo837** [comment]
This should be merged before the first beta release, I think.

**Comment by yoyo837** [comment]
Still need you to do a little new work. https://github.com/ant-design/ant-design/pull/20023#issuecomment-564598709

**Comment by saeedrahimi** [comment]
> 
> 
> Still need you to do a little new work. [#20023 (comment)](https://github.com/ant-design/ant-design/pull/20023#issuecomment-564598709)

I'm working on it ;)

**Comment by saeedrahimi** [comment]
Waiting for [rtl picker to be merged](https://github.com/react-component/picker/pull/6)

**Comment by saeedrahimi** [comment]
Added rtl direction to the new date-picker based on  [this pull request](https://github.com/react-component/picker/pull/6).
Just one more step remaining, after [this pull request](https://github.com/react-component/picker/pull/6) merged and published on NPM, update `package.json`

**Comment by saeedrahimi** [comment]
~ All Green :)

**Comment by zombieJ** [comment]
Test failed. Please help to check if it's by design:

https://circleci.com/gh/ant-design/ant-design/143653?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by zombieJ** [comment]
`4.0-prepare` update the `rc-select` which may change the snapshot. Please merge and reinstall `node_modules`.

**Comment by saeedrahimi** [comment]
> `4.0-prepare` update the `rc-select` which may change the snapshot. Please merge and reinstall `node_modules`.

I've deleted node_modules folder and reinstalled it. there is problem with `rc-dialog` package.
`4.0-prepare` rc-dialog version is `~7.5.2` and snapshots for modal have `div` with `ant-modal-root` class.
but `<div className={'${prefixCls}-root'}>` has been added in `rc-dialog@7.5.14` [link](https://github.com/react-component/dialog/blob/6be81135e6d0bf7ef459280196a03602049d7bb8/src/Dialog.tsx#L434).

To fix, installed `rc-dialog@7.5.14` locally and updated snapshots, after that reverted `rc-dialog` to `@~7.5.2`

**Comment by saeedrahimi** [comment]
All green now,
@yoyo837 @afc163  @zombieJ when this pull request planned to merge?

> I'm done with this pull request and every time base branch changes, the process of resolving conflicts takes long time.

**Comment by yoyo837** [comment]
react-component/tabs/pull/234 block this?

**Comment by saeedrahimi** [comment]
> [react-component/tabs/pull/234](https://github.com/react-component/tabs/pull/234) block this?

It could be fixed with another patch that has less maintenance.

**Comment by zombieJ** [comment]
> > [react-component/tabs/pull/234](https://github.com/react-component/tabs/pull/234) block this?
> 
> It could be fixed with another patch that has less maintenance.

Should [react-component/tabs/pull/234](https://github.com/react-component/tabs/pull/234) merge or it's out of date?

**Comment by saeedrahimi** [comment]
> > > [react-component/tabs/pull/234](https://github.com/react-component/tabs/pull/234) block this?
> > 
> > 
> > It could be fixed with another patch that has less maintenance.
> 
> Should [react-component/tabs/pull/234](https://github.com/react-component/tabs/pull/234) merge or it's out of date?

I've rebased it to make sure it works.
After rebase i tested it with ant-design rtl manually and everything works properly.

**Comment by zombieJ** [comment]
> > > > [react-component/tabs/pull/234](https://github.com/react-component/tabs/pull/234) block this?
> > > 
> > > 
> > > It could be fixed with another patch that has less maintenance.
> > 
> > 
> > Should [react-component/tabs/pull/234](https://github.com/react-component/tabs/pull/234) merge or it's out of date?
> 
> I've rebased it to make sure it works.
> After rebase i tested it with ant-design rtl manually and everything works properly.

`+ rc-tabs@10.0.0-alpha.1`

**Comment by zombieJ** [comment]
hi @saeedrahimi,
conflict need resolve. Is there any block need I help or we can start to merge prepare.

**Comment by yoyo837** [comment]
He has been waiting for the merger, and has been resolving the conflict after each any change from upstream.This time he has not finished.

**Comment by saeedrahimi** [comment]
> hi @saeedrahimi,
> conflict need resolve. Is there any block need I help or we can start to merge prepare.

hi
i'm resolving conflicts. in past 4 hours i rebased twice from `4.0-prepare`.
it takes time to apply 235 commits and resolve their conflicts.

**Comment by zombieJ** [comment]
Note: Hold #20570 until this PR merged.

**Comment by zombieJ** [comment]
@saeedrahimi , ci failed.

File: .github/workflows/deploy-site.yml
Comment (zombieJ) [inline .github/workflows/deploy-site.yml]: This file should not remove

**Comment by zombieJ** [comment]
Some file removed by mistake, should revert back:
* .github/workflows/deploy-site.yml 
* .github/workflows/lighthouse-ci.yml
* .github/workflows/rebase.yml 
* .github/workflows/test.yml

File: components/cascader/index.tsx
Comment (zombieJ) [inline components/cascader/index.tsx]: This typo has been removed in v4

File: components/cascader/index.tsx
Comment (zombieJ) [inline components/cascader/index.tsx]: compatibility code has been removed in v4

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap
Comment (zombieJ) [inline components/config-provider/__tests__/__snapshots__/components.test.js.snap]: Miss prefix className, should check why.

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap
Comment (zombieJ) [inline components/config-provider/__tests__/__snapshots__/components.test.js.snap]: Miss prefix className, should check why.

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap
Comment (zombieJ) [inline components/config-provider/__tests__/__snapshots__/components.test.js.snap]: Miss prefix className, should check why.

File: components/modal/Modal.tsx
Comment (zombieJ) [inline components/modal/Modal.tsx]: this object can merge with upper object

File: components/popover/__tests__/__snapshots__/index.test.js.snap
Comment (zombieJ) [inline components/popover/__tests__/__snapshots__/index.test.js.snap]: Seems strange it only render a empty span without rlt content.

File: components/popover/demo/placement.md
Comment (zombieJ) [inline components/popover/demo/placement.md]: Do not change origin demo.

File: components/steps/index.tsx
Comment (zombieJ) [inline components/steps/index.tsx]: miss origin `className`?

File: components/table/Table.tsx
Comment (zombieJ) [inline components/table/Table.tsx]: `className` is used for inner Table. Not wrapper element.

File: components/tooltip/demo/placement.md
Comment (zombieJ) [inline components/tooltip/demo/placement.md]: Not modify origin demo.

File: package.json
Comment (zombieJ) [inline package.json]: Should use latest config

File: components/popover/demo/placement.md
Comment (saeedrahimi) [reply components/popover/demo/placement.md]: If i remove this style from demo, it will not displays correctly in rtl direction.

antd-dir-rtl-in-demo2

I can handle it by css with demo id like this:
``` less
.ant-row-rtl #components-tooltip-demo-placement {
    direction: ltr;
}
```
Which way do you prefer?

Comment (zombieJ) [reply components/popover/demo/placement.md]: Prefer additional style instead of inline style : )

File: .github/workflows/deploy-site.yml
Comment (saeedrahimi) [reply .github/workflows/deploy-site.yml]: There was an error while pushing with these files, github was denying push on workflow.
After some googling fixed by using PersonalAccessToken instead of password for git account.

File: components/slider/__tests__/__snapshots__/demo.test.js.snap
Comment (zombieJ) [inline components/slider/__tests__/__snapshots__/demo.test.js.snap]: Miss `ant-slider-tooltip`

File: components/dropdown/dropdown.tsx
Comment (zombieJ) [inline components/dropdown/dropdown.tsx]: should pass origin `overlayClassName` here.

**Comment by zombieJ** [comment]
Should remove margin-right which break the Home header line:
截屏2020-01-01下午1 42 50

**Comment by zombieJ** [comment]
Logo should not revert:
截屏2020-01-01下午1 48 41

File: components/dropdown/dropdown.tsx
Comment (saeedrahimi) [reply components/dropdown/dropdown.tsx]: Excuse me, I didn't get it. could you clarify this for me?

Comment (zombieJ) [reply components/dropdown/dropdown.tsx]: OPS. wrong file. It's comments here:
https://github.com/ant-design/ant-design/pull/19380/files#diff-186839a30bf8b9d67a4b10bf7c091d5fR261

File: package.json
Comment (saeedrahimi) [reply package.json]: my rtl classes increased final css size. so if i decrease `maxSize` ci will fail

File: components/slider/__tests__/__snapshots__/demo.test.js.snap
Comment (saeedrahimi) [reply components/slider/__tests__/__snapshots__/demo.test.js.snap]: will be resolved after updating snapshots

File: components/dropdown/dropdown.tsx
Comment (saeedrahimi) [reply components/dropdown/dropdown.tsx]: done

File: components/popover/__tests__/__snapshots__/index.test.js.snap
Comment (saeedrahimi) [reply components/popover/__tests__/__snapshots__/index.test.js.snap]: will be resolved after updating snapshots

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap
Comment (saeedrahimi) [reply components/config-provider/__tests__/__snapshots__/components.test.js.snap]: will be resolved after updating snapshots

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap
Comment (saeedrahimi) [reply components/config-provider/__tests__/__snapshots__/components.test.js.snap]: will be resolved after updating snapshots

File: package.json
Comment (zombieJ) [reply package.json]: 👌

File: components/slider/__tests__/__snapshots__/demo.test.js.snap
Comment (saeedrahimi) [reply components/slider/__tests__/__snapshots__/demo.test.js.snap]: strange, not resolved 🤔

File: components/config-provider/__tests__/__snapshots__/components.test.js.snap
Comment (saeedrahimi) [reply components/config-provider/__tests__/__snapshots__/components.test.js.snap]: not resolved 🤔

Comment (saeedrahimi) [reply components/config-provider/__tests__/__snapshots__/components.test.js.snap]: that was a mistake, i was looking on outdated changes!

File: components/cascader/index.tsx
Comment (zombieJ) [inline components/cascader/index.tsx]: missing origin `popupClassName`.

File: components/menu/index.tsx
Comment (zombieJ) [inline components/menu/index.tsx]: Is that possible to switch `MenuContext.Provider` with `ConfigConsumer` to let `direction` from `ConfigConsumer` to the `MenuContext.Provider`? Since use `this. direction` may failed in concurrent mode.

File: components/select/index.tsx
Comment (zombieJ) [inline components/select/index.tsx]: should pass origin `dropdownClassName`.

**Comment by HDaghash** [comment]
What a great effort @saeedrahimi !,
@zombieJ  many thanks for keep following on this PR I think this should be there ASAP.

**Comment by Memadaldin** [comment]
@saeedrahimi Amazing work!

**Comment by zombieJ** [comment]
Header button background should be transparent:
截屏2020-01-02上午11 26 18

**Comment by ycjcl868** [comment]
@saeedrahimi `ConfigProvider` RTL demo border-color should be `#303030`, you can add

```css
<style>
[data-theme="theme"] .direction-components {
  border-bottom: 1px solid #303030;
}
</style>
```


image

**Review by ycjcl868 — APPROVED** [review]

**Comment by shayegh** [comment]
Hi,
There is an issue in the menu component style. the border of the menu must be at the left side.
please check this one

**Comment by zombieJ** [comment]
@saeedrahimi, 
Since to avoid continue conflict, some PR is pending on this PR. I will continue work with your branch by commits. If push conflict, please pull firstly to avoid our code conflict.

**Comment by zombieJ** [comment]
@saeedrahimi, 
Since to avoid continue conflict, some PR is pending on this PR. I will continue work with your branch by commits. If push conflict, please pull firstly to avoid our code conflict.

**Comment by zombieJ** [comment]
> @saeedrahimi,
> Since to avoid continue conflict, some PR is pending on this PR. I will continue work with your branch by commits. If push conflict, please pull firstly to avoid our code conflict.

To avoid conflict, I will use `merge` instead of `rebase`.

**Comment by saeedrahimi** [comment]
> Hi,
> There is an issue in the menu component style. the border of the menu must be at the left side.
> please check this one
> menu

border is in the left side in `ltr` direction. so it should be in right side in `rtl` direction.
border should not be outside of sidebar

**Comment by saeedrahimi** [comment]
Thank you @zombieJ for fixing my mistakes 👍

**Comment by yoyo837** [comment]
Amazing works！

**Comment by vahidalvandi** [comment]
thank you how i can use this in nuxt‌?

**Comment by afc163** [comment]
image

Is this header misordered?

**Comment by saeedrahimi** [comment]
> image
> 
> Is this header misordered?

I think @zombieJ  fixed it in this [commit](https://github.com/ant-design/ant-design/commit/dafc5a2f1d1fb06901231fba9f45f588a400d40d) cause i see header in correct order in [next.ant.design](https://next.ant.design)

**Comment by afc163** [comment]
Could you help to fix it to correct order?

**Comment by saeedrahimi** [comment]
> Could you help to fix it to correct order?

Is there any misordered item? 🤔
antd-header

 ZombieJ fixed that.

**Comment by zombieJ** [comment]
That's fine. I've resolved.

**Comment by PejmanNik** [comment]
Amazing work @vahidalvandi ❤
I just find a bug, the `ant-col-rtl` broken all `:not([class^='ant-col-']):not([class*=' ant-col-'])` selectors

**Comment by yoyo837** [comment]
@PejmanNik Feel free to open a new issue about that.

**Comment by afsanefda** [comment]
Hi all,
I have a problem with ant Tabs! **onclick** is not showing the content. **The arrow buttons** are not in the correct position. Here is my code and the implementation result!

```
<ConfigProvider direction="rtl" locale="fa">

        <div className={style.body}>
          <Tabs
            className={style.tab}
            defaultActiveKey="0"
            tabPosition="top"
            style={{ height: 220 }}
          >
            {[...Array(30).keys()].map((i) => (
              <TabPane tab={`تب${i}`} key={i}>
              محتویات
                {i}
              </TabPane>
            ))}
          </Tabs>
        </div>
      </ConfigProvider>
```

Screenshot (3)

It would be great if you could help!

**Comment by afsanefda** [comment]
Is this working with **Tabs**? Here is my code:

```
<ConfigProvider direction="rtl">
          <Tabs
            className={style.tab}
            defaultActiveKey="0"
            tabPosition="top"
            style={{ height: 220 }}
          >
            {[...Array(30).keys()].map((i) => (
              <TabPane tab={`تب${i}`} key={i}>
                محتویات
                {i}
              </TabPane>
            ))}
          </Tabs>
  </ConfigProvider>
```
and the result is:

Screenshot (4)

there is no change for tabs!

**Comment by adirrapidapi** [comment]
how to make Sider in right side (rtl)?

**Comment by xrkffgg** [comment]
> how to make Sider in right side (rtl)?

You can use https://ant.design/components/config-provider-cn/#components-config-provider-demo-direction to set rtl, so the base Slider can be in right side.
or https://ant.design/components/slider-cn/#components-slider-demo-reverse in ltr

**Comment by adirrapidapi** [comment]
@xrkffgg  I added with configProvider the rtl direction but the sider it's still in left side, as you can see in this example [https://stackblitz.com/edit/react-ouxpra-trwwbf?embed=1&file=index.js](https://stackblitz.com/edit/react-ouxpra-trwwbf?embed=1&file=index.js)

**Comment by adirrapidapi** [comment]
@xrkffgg any solution?

**Comment by xrkffgg** [comment]
> @xrkffgg any solution?

sorry, I just saw. I'll fix it.

**Comment by mehdiraized** [comment]
@saeedrahimi i have issue on nextjs
```
Warning: Prop `className` did not match. Server: "ant-badge ant-badge-rtl" Client: "ant-badge"
```

**Comment by xrkffgg** [comment]
@mehdiraized Please create new issue https://github.com/ant-design/ant-design/issues/new/choose

**Pull request merged**
